core/controllers/reader.py

```diff
@@ -731,16 +731,67 @@ class StateCompleteEventHandler(base.BaseHandler):
 
     REQUIRE_PAYLOAD_CSRF_CHECK = False
 
+    URL_PATH_ARGS_SCHEMAS = {
+         'exploration_id': {
+            'schema': editor.SCHEMA_FOR_EXPLORATION_ID
+        }
+    }
+
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'state_name': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'has_length_at_most',
+                        'max_value': constants.MAX_STATE_NAME_LENGTH
+                    }]
+                }
+            },
+             'session_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'time_spent_in_state_secs': {
+                'schema': {
+                    'type': 'float',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        'min_value': 0
+                    }]
+                }
+            },
+            'exp_version': {
+                'schema': {
+                    'type': 'int',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        # Version must be greater than zero.
+                        'min_value': 1
+                    }]
+                },
+                'default_value': None
```

Comment:
Why can this be `None`?

Reply:
Similar to the other comment. the test ```test_cannot_complete_state_with_no_exploration_version``` fails when it does NOT default to ```None``` due to an assertion to a specific error message in the test when ```exp_version``` is  ```None```.  Not sure if it is valid that it should default to ```None``` though.

Reply:
After going through the frontend. This handler is only called in the function ```recordStateCompletedAsync``` with all arguments. I.E never with ```Null``` or ```None```. So why ```exp_version``` needs to be nullable for the backend-test ```test_cannot_complete_state_with_no_exploration_version``` confuses me. Not sure what to do going forward.

Reply:
Just remove the test that passes `None` into the exp ID. And then disallow `None`.

Reply:
Done! @vojtechjelinek